### ISSUE TITLE
resy returns image width of pixmap

### ISSUE THREAD

**Issue by paaguti** [pymupdf/PyMuPDF#3023]
### Description of the bug

When extracting a pixmap, the resy field returns the width(!) and not the Y resolution

### How to reproduce the bug

This is the test program I have written and tested on different PDFs with images:

```
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
'''Just a test'''

from argparse import ArgumentParser
from argparse import RawDescriptionHelpFormatter
import fitz

def main():
    parser = ArgumentParser(description=__doc__,
                            formatter_class=RawDescriptionHelpFormatter)

    parser.add_argument('pdf', metavar='PDF',
                        type=str, help='The PDF file')
    parser.add_argument('page', metavar='PAGE',
                        type=int, help='The page')
    args = parser.parse_args()

    print(f'fitz version: {fitz.version}')
    with fitz.open(args.pdf) as doc:
        assert (
            args.page in range(doc.page_count)
        ), f"Page {args.page} out of range, PDF has {doc.page_count} pages!"
        images = doc.get_page_images(args.page-1)
        for image in images:
            pixmap = fitz.Pixmap(doc, image[0])
            print(f'(height: {pixmap.height:5d}, width: {pixmap.width}')
            print(f'(resx:   {pixmap.xres:5d}, resy:  {pixmap.yres}')
            print(f' Colorspace [{pixmap.colorspace}]')

if __name__ == '__main__':
    main()
```

These are the results:

```
paag@840g5  ~  pdf-test Downloads/Commentarii_de_bello_civili.pdf 5    
fitz version: ('1.23.9', '1.23.9', '20240111000001')
(height:  4650, width: 2892
(resx:      96, resy:  2892
 Colorspace [Colorspace(CS_RGB) - DeviceRGB]
(height:   204, width: 1034
(resx:      96, resy:  1034
 Colorspace [Colorspace(CS_RGB) - DeviceRGB]
 paag@840g5  ~  pdf-test Downloads/ma_Lareliquia_1902.pdf 10
fitx version: ('1.23.9', '1.23.9', '20240111000001')
(height:   385, width: 227
(resx:      96, resy:  227
 Colorspace [Colorspace(CS_RGB) - DeviceRGB]
(height:   385, width: 227
(resx:      96, resy:  227
 Colorspace [Colorspace(CS_RGB) - DeviceRGB]
(height:  2141, width: 1259
(resx:      96, resy:  1259
 Colorspace [Colorspace(CS_RGB) - DeviceRGB]
 paag@840g5  ~  pdf-test Downloads/ma_ElcrimendelPadreAmaro_1901.pdf 10 
fitz version: ('1.23.9', '1.23.9', '20240111000001')
(height:   373, width: 227
(resx:      96, resy:  227
 Colorspace [Colorspace(CS_RGB) - DeviceRGB]
(height:   373, width: 227
(resx:      96, resy:  227
 Colorspace [Colorspace(CS_RGB) - DeviceRGB]
(height:  2070, width: 1261
(resx:      96, resy:  1261
 Colorspace [Colorspace(CS_RGB) - DeviceRGB]

```

### PyMuPDF version

1.23.9

### Operating system

Linux

### Python version

3.10

**Comment by julian-smith-artifex-com** [comment]
Thanks for this. I have a fix in my tree.

We're planning to make a release later today with the fix.

**Comment by julian-smith-artifex-com** [comment]
Fixed in PyMuPDF-1.23.11.
